.travis.yml

```diff
@@ -0,0 +1,85 @@
+sudo: required
+dist: trusty
+
+os: linux
+
+language: c
+
+compiler:
+  - gcc
+#  - clang
+
+addons:
+  apt:
+    packages:
+    - mysql-server
+    
+cache:
+  directories:
+  - download-cache
+
+env:
+  global:
+    - LUAJIT_LIB=/usr/lib64/libluajit-5.1.so
+    - LUAJIT_INC=/usr/include/luajit-2.0
+    - LUA_INCLUDE_DIR=/usr/include/luajit-2.0
+    - LUA_CMODULE_DIR=/lib
+  matrix:
+    - NGINX_VERSION=1.9.15
+#    - NGINX_VERSION=1.10.0
+
+services: 
+ - memcache
+ - redis-server
+    
+before_install:
+  - sudo apt-get install -y axel cpanminus libluajit-5.1-dev libgd-dev
+  
+install:
+  - if [ ! -d download-cache ]; then mkdir download-cache; fi
+  - if [ ! -f download-cache/drizzle7-2011.07.21.tar.gz ]; then wget -O download-cache/drizzle7-2011.07.21.tar.gz http://openresty.org/download/drizzle7-2011.07.21.tar.gz; fi
+  - wget http://ftp.cs.stanford.edu/pub/exim/pcre/pcre-8.33.tar.gz
+  - git clone https://github.com/openresty/test-nginx.git
+  - git clone https://github.com/openresty/openresty.git ../openresty
+  - git clone https://github.com/openresty/no-pool-nginx.git ../no-pool-nginx
+  - git clone https://github.com/PiotrSikora/ngx_http_auth_request_module.git ~/work/nginx/ngx_http_auth_request_module-0.2
+  - git clone https://github.com/openresty/nginx-devel-utils.git
+  - git clone https://github.com/openresty/mockeagain.git
+  - git clone https://github.com/openresty/lua-cjson.git
+  - git clone https://github.com/openresty/lua-upstream-nginx-module.git ../lua-upstream-nginx-module
+  - git clone https://github.com/openresty/echo-nginx-module.git ../echo-nginx-module
+  - git clone https://github.com/openresty/nginx-eval-module.git ../nginx-eval-module
+  - git clone https://github.com/simpl/ngx_devel_kit.git ../ndk-nginx-module
+  - git clone https://github.com/FRiCKLE/ngx_coolkit.git ../coolkit-nginx-module
+  - git clone https://github.com/openresty/headers-more-nginx-module.git ../headers-more-nginx-module
+  - git clone https://github.com/openresty/drizzle-nginx-module.git ../drizzle-nginx-module
+  - git clone https://github.com/openresty/set-misc-nginx-module.git ../set-misc-nginx-module
+  - git clone https://github.com/openresty/memc-nginx-module.git ../memc-nginx-module
+  - git clone https://github.com/openresty/rds-json-nginx-module.git ../rds-json-nginx-module
+  - git clone https://github.com/openresty/srcache-nginx-module.git ../srcache-nginx-module
+  - git clone https://github.com/openresty/redis2-nginx-module.git ../redis2-nginx-module
+  
+before_script:
+  - mysql -uroot -e 'create database ngx_test; grant all on ngx_test.* to "ngx_test"@"%" identified by "ngx_test"; flush privileges;'
+  
+script:
+  - tar xzf download-cache/drizzle7-2011.07.21.tar.gz && cd drizzle7-2011.07.21 
+  - ./configure --prefix=/usr --without-server > build.log 2>&1 || cat build.log
+  - sudo make libdrizzle-1.0 install-libdrizzle-1.0 > build.log 2>&1 || cat build.log
+  - cd ..
+  - cd mockeagain && make && cd ..
```

Comment:
pardon ?

[ilia@localhost t]$ grep -i mock *
grep: 023-rewrite: Is a directory
grep: 024-access: Is a directory
057-flush-timeout.t:    if ($ENV{LD_PRELOAD} !~ /\bmockeagain\.so\b/) {
057-flush-timeout.t:        $ENV{LD_PRELOAD} = "mockeagain.so $ENV{LD_PRELOAD}";
057-flush-timeout.t:    if ($ENV{MOCKEAGAIN} eq 'r') {
057-flush-timeout.t:        $ENV{MOCKEAGAIN} = 'rw';
057-flush-timeout.t:        $ENV{MOCKEAGAIN} = 'w';
057-flush-timeout.t:    $ENV{MOCKEAGAIN_WRITE_TIMEOUT_PATTERN} = 'hello, world';
065-tcp-socket-timeout.t:    if ($ENV{LD_PRELOAD} !~ /\bmockeagain\.so\b/) {
065-tcp-socket-timeout.t:        $ENV{LD_PRELOAD} = "mockeagain.so $ENV{LD_PRELOAD}";
065-tcp-socket-timeout.t:    if ($ENV{MOCKEAGAIN} eq 'r') {
065-tcp-socket-timeout.t:        $ENV{MOCKEAGAIN} = 'rw';
065-tcp-socket-timeout.t:        $ENV{MOCKEAGAIN} = 'w';
065-tcp-socket-timeout.t:    $ENV{MOCKEAGAIN_WRITE_TIMEOUT_PATTERN} = 'get helloworld';
117-raw-req-socket-timeout.t:    if ($ENV{LD_PRELOAD} !~ /\bmockeagain\.so\b/) {
117-raw-req-socket-timeout.t:        $ENV{LD_PRELOAD} = "mockeagain.so $ENV{LD_PRELOAD}";
117-raw-req-socket-timeout.t:    if ($ENV{MOCKEAGAIN} eq 'r') {
117-raw-req-socket-timeout.t:        $ENV{MOCKEAGAIN} = 'rw';
117-raw-req-socket-timeout.t:        $ENV{MOCKEAGAIN} = 'w';
117-raw-req-socket-timeout.t:    $ENV{MOCKEAGAIN_WRITE_TIMEOUT_PATTERN} = 'hello, world';
131-duplex-req-socket.t:    if ($ENV{LD_PRELOAD} !~ /\bmockeagain\.so\b/) {
131-duplex-req-socket.t:        $ENV{LD_PRELOAD} = "mockeagain.so $ENV{LD_PRELOAD}";
131-duplex-req-socket.t:    if ($ENV{MOCKEAGAIN} eq 'r') {
131-duplex-req-socket.t:        $ENV{MOCKEAGAIN} = 'rw';
131-duplex-req-socket.t:        $ENV{MOCKEAGAIN} = 'w';
131-duplex-req-socket.t:    $ENV{MOCKEAGAIN_WRITE_TIMEOUT_PATTERN} = 'slow';
grep: cert: Is a directory
grep: data: Is a directory
grep: lib: Is a directory
[ilia@localhost t]$